### PR TITLE
Add FileEntry::linkto for symbolic links

### PR DESCRIPTION
**Pull request by marxin**

### CODE REVIEW

File: src/rpm/package.rs [hunk 1]
```diff
@@ -843,8 +843,13 @@ impl PackageMetadata {
             Err(Error::TagNotFound(_)) => Ok(None),
             Err(e) => return Err(e),
         };
+        let links = self
+            .header
+            .get_entry_data_as_string_array(IndexTag::RPMTAG_FILELINKTOS);
```
Comment (dralley) [inline src/rpm/package.rs:848]: Presuming an RPM has no files with symlinks whatsoever, is `RPMTAG_FILELINKTOS` guaranteed to exist?  If yes, this is fine, if no, then it should be treated like `RPMTAG_FILECAPS` a few lines above so that we don't erroneously return a bad result.

File: src/tests.rs [hunk 2]
```diff
@@ -333,6 +333,18 @@ fn test_rpm_header() -> Result<(), Box<dyn std::error::Error>> {
 
     // Verify that if there are no capabilities set then the caps field is None
```
Comment (dralley) [inline src/tests.rs:334]: We might want to have a negative test for linktos similar to how this test is a negative test for capabilities.

File: src/rpm/package.rs (hunk 1 above)
Comment (marxin) [reply src/rpm/package.rs:848]: Don't know based on the documentation: https://rpm-software-management.github.io/rpm/manual/tags.html. However, I've just added a new symlink-free test-case and the tag is present. Hope it helps.

File: src/tests.rs (hunk 2 above)
Comment (marxin) [reply src/tests.rs:338]: Yep. I've improved the assert macro usage where it expected an empty string for non-symlink entires and a non-empty one for a symlink.

File: src/rpm/package.rs (hunk 1 above)
Comment (dralley) [reply src/rpm/package.rs:848]: It looks like it adds the tag if there are files (regardless if any are symlinks), and always does if there are, and doesn't otherwise.  But in the event that there are no files then we should return no file entries, so it's fine.  

TL;DR this is OK, so long as there aren't RPMs floating around with different behavior.

**Review by dralley — APPROVED** [review]
